utils/misc/retry.go

```diff
@@ -0,0 +1,30 @@
+package misc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+func RetryWith(parentContext context.Context, timeout time.Duration, maxAttempts int, f func(ctx context.Context) error) error {
```

Comment:
Add unit test scenarios

**- Scenario 1**: no timeout -> no error after 1 attempt
**- Scenario 2**: timeout after maxAttempts -> error after maxAttempts
**- Scenario 3**: parentContext timeout -> error after 1 attempt
**- Scenario 4**: timeout on second attempt -> no error after 2 attempt

Reply:
test added for all the above four scenarios